File: dotnet/DesignPatterns/StructuralDesignPatterns/BridgeDesignPattern/BridgeDesignPattern.Example1/Radio.cs
namespace BridgeDesignPattern.Example1;

public class Radio : IDevice
{
	public bool IsEnabled { get; private set; }
	public int Volume { get; set; } = 5;
	public int Channel { get; set; } = 88;
	public void Enable()
	{
		IsEnabled = true;
	}
	public void Disable()
	{
		IsEnabled = false;
	}
}

File: java/DesignPatterns/StructuralDesignPatterns/BridgeDesignPattern/BridgeDesignPattern.Example1/src/main/java/structural/bridge/example1/Radio.java
package structural.bridge.example1;

public class Radio implements IDevice {
    private boolean enabled;
    private int volume = 5;
    private int channel = 88;

    @Override public boolean isEnabled() { return enabled; }
    @Override public int getVolume() { return volume; }
    @Override public void setVolume(int volume) { this.volume = volume; }
    @Override public int getChannel() { return channel; }
    @Override public void setChannel(int channel) { this.channel = channel; }
    @Override public void enable() { enabled = true; }
    @Override public void disable() { enabled = false; }
}
